### PR TITLE
Implement [MH3] Lethal Throwdown

### PR DESCRIPTION
**Pull request by grimreap124**
[[Lethal Throwdown]]

### CODE REVIEW

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * @author grimreap124
+ */
+public final class LethalThrowdown extends CardImpl {
+
+    public LethalThrowdown(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.SORCERY }, "{B}");
+
+        // As an additional cost to cast this spell, sacrifice a creature or sacrifice a modified creature.
+        this.getSpellAbility().addCost(new SacrificeTargetCost(StaticFilters.FILTER_PERMANENT_CREATURE)
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: This is not correct. It is worded that way in order to allow the player to not draw if it is detrimental.

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    private LethalThrowdownEffect(final LethalThrowdownEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent sacrificedPermanent = null;
+        boolean isModified = false;
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/l/LethalThrowdown.java:66]: We have a ModifiedPredicate (I think that's its actual name. Look at other modified cards implementation)

**Review by Susucre — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java (hunk 1 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: What part of it isn't right? Don't they have to sacrifice either a creature or a modified creature to cast the spell?

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java (hunk 2 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:66]: I looked at a lot of them but there was no "do something additional if targeted creature is modified" they are more X where X is amount of modified creatures. 

I saw the Predicate but couldn't figure out how to check if a permanent fulfills it

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java (hunk 1 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: There is two alternatives to the additional cost. It is possible to sacrifice a modified creature to the first option (sacrifice a creature), in which case there is no card draw.

Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: I handle the alternative of the creature being modified in LethalThrowdownEffect.
But maybe I'm not understanding the problem

Comment (Susucre) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: On casting the spell, the player should have the choice between 'sacrifice a creature' and 'sacrifice a modified creature'. Then pay for that choice. A modified creature could be sacrificed for either.

On resolve, the draw a card part will only applied if the 'sacrificed a modified creature' cost has been paid.

Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java:35]: I see what you're saying, I'll update it soon

Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java]: I have pushed an update, I don't know how much you'll like it though. I couldn't find a way to get the selected cost of an OrCost and then nicely compare it to the different costs so I just used getText()

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            if (cost instanceof OrCost) {
+                OrCost orCost = (OrCost) cost;
+                if (orCost.getSelectedCost().getText().equals("sacrifice a modified creature")) {
+                    new DrawCardSourceControllerEffect(1).apply(game, source);
+                    break;
+                }
+            }
+        }
+
+        DestroyTargetEffect effect = new DestroyTargetEffect();
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/l/LethalThrowdown.java:80]: Effects should apply in the order of the text.
Anyway will be irrelevant if you only apply the "If the modified creature was sacrificed, draw a card" part in the custom effect, and add the destroy effect to the card's ability directly.

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public LethalThrowdown(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.SORCERY }, "{B}");
+
+        // As an additional cost to cast this spell, sacrifice a creature or sacrifice a modified creature.
+        this.getSpellAbility()
+                .addCost(new OrCost("sacrifice a creature or sacrifice a modified creature",
+                        new SacrificeTargetCost(StaticFilters.FILTER_PERMANENT_CREATURE)
+                                .setText("sacrifice a creature"),
+                        new SacrificeTargetCost(filter).setText("sacrifice a modified creature")));
+        // Destroy target creature or planeswalker. If the modified creature was sacrificed, draw a card.
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/l/LethalThrowdown.java:42]: Add the Destroy effect with `this.getSpellAbility().addEffect(new DestroyTargetEffect());` here. No need for it to be in the custom effect.

File: Mage/src/main/java/mage/abilities/costs/OrCost.java [hunk 5]
```diff
@@ -90,8 +94,7 @@ public boolean pay(Ability ability, Game game, Ability source, UUID controllerId
                         Outcome.Detriment, sb.toString(), null,
                         CardUtil.getTextWithFirstCharUpperCase(usable.get(0).getText()),
                         CardUtil.getTextWithFirstCharUpperCase(usable.get(1).getText()),
-                        ability, game
-                )) {
+                        ability, game)) {
```
Comment (Susucre) [inline Mage/src/main/java/mage/abilities/costs/OrCost.java:97]: There is a risk for heavy changes in big files if your IDE is auto formatting those kind of line breaks.

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    private LethalThrowdownEffect(final LethalThrowdownEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        for (Cost cost : source.getCosts()) {
+            if (cost instanceof OrCost) {
+                OrCost orCost = (OrCost) cost;
+                if (orCost.getSelectedCost().getText().equals("sacrifice a modified creature")) {
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/l/LethalThrowdown.java:73]: I don't have a more elegant solution, and I think this is fine.

Could you add a comment though if someone encounters the effect later on, to understand it is an ad hoc solution to a specific issue for this card.

File: Mage/src/main/java/mage/abilities/costs/OrCost.java (hunk 5 above)
Comment (Susucre) [reply Mage/src/main/java/mage/abilities/costs/OrCost.java:97]: I'm no Vscode config expert, but aparently you can format only on modifications and not whole file by adding that to your config file:
`"editor.formatOnSaveMode": "modifications"`

Comment (grimreap124) [reply Mage/src/main/java/mage/abilities/costs/OrCost.java:97]: Thanks, I've changed to that hopefully it helps.

File: Mage.Sets/src/mage/cards/l/LethalThrowdown.java (hunk 3 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/l/LethalThrowdown.java]: I noticed that when I pushed it but got distracted and forgot to fix it but I have moved it now anyway

**Review by Susucre — APPROVED** [review]

**Pull request merged**